pennylane/ops/functions/eigvals.py

```diff
@@ -14,20 +14,31 @@
 """
 This module contains the qml.eigvals function.
 """
+import warnings
+
 # pylint: disable=protected-access
 from functools import reduce
-import warnings
+
+import scipy
 
 import pennylane as qml
 
 
 @qml.op_transform
-def eigvals(op):
+def eigvals(op, k=1, which="SA"):
     r"""The eigenvalues of one or more operations.
 
+    For a SparseHamiltonian object, the eigenvalues are computed with the efficient ``scipy.sparse.linalg.eigsh``
+    method which returns `k` eigenvalues. For an :math:`N \times N` sparse matrix, `k` must be smaller than `N - 1`,
+    otherwise ``scipy.sparse.linalg.eigsh`` fails. If the requested `k` is equal or larger than `N - 1`,
+    the regular ``qml.math.linalg.eigvalsh`` is applied on the dense matrix.
+
     Args:
         op (.Operator, pennylane.QNode, .QuantumTape, or Callable): An operator, quantum node, tape,
             or function that applies quantum operations.
+        k (int): The number of eigenvalues and eigenvectors to be returned for a SparseHamiltonian.
+        which (str): Method for computing the eigenvalues of a SparseHamiltonian with Scipy, possible values are
+            ["LM" | "SM" | "LA" | "SA" | "BE"]

```

Comment:
What do these stand for? I personally have no idea. I suggest quickly explaining what each of these mean, e.g.,
```
Possible values are "LM" (linear method), "SM" (sparse method), etc
```

Reply:
Could also add link to scipy docs